### PR TITLE
v0.1

### PR DESCRIPTION
**Pull request by natebrennand**
Thoughts? Don't hold back

---

[Query Building](https://github.com/natebrennand/data.adicu.com/blob/18d036a1bd59ac54ae29d129020a2ace3f324d5e/data/lib/query.py)

Not convinced this is the best way to go about it but seems more manageable than the prior setup. Requires declaring [a dict like this](https://github.com/natebrennand/data.adicu.com/blob/18d036a1bd59ac54ae29d129020a2ace3f324d5e/data/housing/housing.py#L22-L79) for each table.

---

[Errors](https://github.com/natebrennand/data.adicu.com/blob/18d036a1bd59ac54ae29d129020a2ace3f324d5e/data/errors/errors.py)

Handled with a decorator on routes. When an AppError Exception is raised the decorator catches it and constructs an error to be returned. We'll define errors in a JSON which we can then use simultaneously for documentation.

---

[Testing](https://github.com/natebrennand/data.adicu.com/tree/18d036a1bd59ac54ae29d129020a2ace3f324d5e/data/tests)

We'll use a [standard limited db dump](https://github.com/natebrennand/data.adicu.com/tree/18d036a1bd59ac54ae29d129020a2ace3f324d5e/dev) for testing so that we can test the responses consistently. It's not complete for courses v2 yet but will work for housing and courses v1.

### CODE REVIEW

**Comment by rbtying** [comment]
What makes the new dict better than the old dict...? My apirefactor a little while back introduced configuration dicts, and it seems awfully similar at the moment.

**Comment by znewman01** [comment]
@rbtying i don't think it's any better/worse than the old config dicts, but the from-scratch rewrite is nice

**Comment by znewman01** [comment]
procfile is unnecessary IMO

**Comment by natebrennand** [comment]
@rbtying, it's a similar pattern but it seems like you were setting configuration on a per-table basis. The main focus was moving to flask from tornado

**Comment by natebrennand** [comment]
The build works with Travis CI!

File: data/lib/query.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    statements = []
+    values = []
+    # iterate over the querystring params
+    for attr, val in request.args.iteritems():
+        try:
+            statements.append(attr_converter[attr]['converter'](
+                attr_converter[attr]['column']))
+            values.append(val)  # add after the possible keyerror
+        except KeyError:
+            # TODO: handle invalid params somehow or continue to ignore
```
Comment (brishin) [inline data/lib/query.py]: Can an AppError be thrown here?

File: data/housing/housing.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        raise errors.AppError('NO_RESULTS')
+    return make_response(json.dumps({
+        'results': results,
+        'status': 200
+    }), 200)
+
+
+@housing.route('/rooms')
+@housing.route('/rooms/<int:page>')
+@errors.catch_error
```
Comment (brishin) [inline data/housing/housing.py]: Can you use
http://flask.pocoo.org/docs/api/#flask.Flask.errorhandler
instead of decorating each function?

**Comment by brishin** [comment]
Looking good. :+1: 

It looks like `scripts/` need to be updated to run properly.
`data/scripts/pg_es_import.py` still has a dependency on tornado (specifically `tornado.httpclient`)
I wonder if there are other things that still rely on tornado.

File: data/housing/housing.py (hunk 2 above)
Comment (natebrennand) [inline data/housing/housing.py]: You're right, it's much cleaner

https://github.com/natebrennand/data.adicu.com/commit/3fb342e982f02fefe270efd928d2cdb251118520

File: data/lib/query.py (hunk 1 above)
Comment (natebrennand) [inline data/lib/query.py]: added it and made the messages generated based on the incorrect attribute

**Comment by brishin** [comment]
:ok: 
lgtm. Will merge into the v2 branch if there are no more comments in 24h.

**Comment by natebrennand** [comment]
additional changes:
- just cleaned up the error handling
- changed the format of the results so the keys match the attributes in the queries

**Pull request merged**